### ISSUE TITLE
hasRoute not working properly

### ISSUE THREAD

**Issue by ritikrao1** [fastify/fastify#5503]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.26.2

### Node.js version

14.18.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

14.3

### Description

hasRoute is giving me false where as the route with same http method and url exists

reproduce from here 
```const fastify = require("fastify");
const app = fastify({logger: true, caseSensitive: false});
app.register(function (app,options, done){
    app.route({
        method: "POST",
        url : '/delivery-service/cancellation/getpafcancellationdetails', 
        handler: async (req, res) => {
            console.log("in api handler")
            req.result = { msg: "sent successfully"};
            res.status(200).send(req.result);
        }
    });
    done();
})



app.register(function (app, options ,done){
    app.hasRoute({ method: "post", url: "/delivery-service/cancellation/getpafcancellationdetails"})? null :
    app.route({
        method: "POST",
        url : '/delivery-service/cancellation/getpafcancellationdetails', 
        handler: async (req, res) => {
            console.log("in api handler")
            req.result = { msg: "sent successfully"};
            res.status(200).send(req.result);
        }
    });
    done();
})

app.addHook('onReady', (done) => {
    console.log("Service running on port: 3000");
    done();
});
app.listen({host: 'localhost', port: 3000});
```

### Expected Behavior

hasRoute should return true in when i call it in second app register, as the api with that url and method exists

**Comment by matthyk** [comment]
This is correct behaviour since the route only exists in the first context but not in the second one but you call the `hasRoute` method within the second context.

**Comment by ritikrao1** [comment]
but then how would we know if a route exists in other context or not, because registering same route here gives error

**Comment by mcollina** [comment]
At first glance, I think we are not validating that `"post" === "POST"`. Can you try using the capitalize method when calling `hasRoute`?

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by codershiba** [comment]
> Would you like to send a Pull Request to address this issue? Remember to add unit tests.

What is the intended behaviour here? Should lowercase `post` be allowed here or not?

**Comment by jsumners** [comment]
The check should be case insensitive.

**Comment by SMNBLMRR** [comment]
I have submitted a PR that should fix issue #5508  I hope I did everything correctly. I apologize if there are any mistakes, but this is my first PR 😃
